### PR TITLE
allow setting limit when fetching projects

### PR DESCRIPTION
**Pull request by akphi**
Allow setting limit when fetching projects (/projects endpoint). This would be useful to make the API typeahead friendly, for more context, see https://github.com/finos/legend-studio/issues/408

### CODE REVIEW

File: legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabProjectApi.java [hunk 1]
```diff
@@ -143,7 +143,27 @@ else if (types instanceof Set)
...
+                    {
+                        // If the number of projects found already exceed the limit, skip the check for the other modes
+                        break;
+                    }
+                }
+            }
+            // ensure the list of returned projects cannot exceed the limit (if specified) for whatever reasons
+            if (limit != null)
+            {
+                return projects.stream().limit(limit).collect(Collectors.toList());
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabProjectApi.java:166]:
```suggestion
                while (projects.size() > limit)
                {
                    projects.remove(projects.size() - 1);
                };
```

**Review by kevin-m-knight-gs — CHANGES_REQUESTED** [review]

File: legend-sdlc-server/src/test/java/org/finos/legend/sdlc/server/inmemory/backend/api/InMemoryProjectApi.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public Project getProject(String id)
     }
 
     @Override
-    public List<Project> getProjects(boolean user, String search, Iterable<String> tags, Iterable<ProjectType> types)
+    public List<Project> getProjects(boolean user, String search, Iterable<String> tags, Iterable<ProjectType> types, Integer limit)
```
[the remark this answers is not in the document]
Comment (akphi) [reply legend-sdlc-server/src/test/java/org/finos/legend/sdlc/server/inmemory/backend/api/InMemoryProjectApi.java:46]: @kevin-m-knight-gs but in the original implementation, we don't take into account `tag`, `search`, etc.

[the remark this answers is not in the document]
Comment (akphi) [reply legend-sdlc-server/src/test/java/org/finos/legend/sdlc/server/inmemory/backend/api/InMemoryProjectApi.java:52]: done!

File: legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabProjectApi.java (hunk 1 above)
Comment (akphi) [reply legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabProjectApi.java]: as discussed, now we use `subList` to not create an extra list. Thanks!

**Pull request merged**